Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/TvUtil.java

```diff
@@ -0,0 +1,278 @@
+				context.grantUriPermission(LEANBACK_PACKAGE, contentUri, FLAG_GRANT_READ_URI_PERMISSION);
+			}
+			catch (Exception e)
+			{
+				Log.e(TAG, "Failed to create banner");
+				Log.e(TAG, e.getMessage());
+			}
+		}
+		return contentUri;
+	}
+
+	private static String buildBannerFilename(String gameId)
+	{
+		return Environment.getExternalStorageDirectory().getPath() +
+			"/dolphin-emu/Cache/" + gameId + "_banner.png";
+	}
+
+	/**
+	 * Needed since debug builds append '.debug' to the end of the package
+	 */
+	private static String getFilePrivider(Context context)
+	{
+		return context.getPackageName() + ".filesprovider";
+	}
+
+	/**
+	 * Schedules syncing channels via a {@link JobScheduler}.
+	 *
+	 * @param context for accessing the {@link JobScheduler}.
+	 */
+	public static void scheduleSyncingChannel(Context context)
+	{
+		ComponentName componentName = new ComponentName(context, SyncChannelJobService.class);
+		JobInfo.Builder builder = new JobInfo.Builder(1, componentName);
+		builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY);
+
+		JobScheduler scheduler =
+			(JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
+
+		Log.d(TAG, "Scheduled channel creation.");
+		scheduler.schedule(builder.build());
+	}
+
+	/**
+	 * Schedulers syncing programs for a channel. The scheduler will listen to a {@link Uri} for a
+	 * particular channel.
+	 *
+	 * @param context   for accessing the {@link JobScheduler}.
+	 * @param channelId for the channel to listen for changes.
+	 */
+	@TargetApi(Build.VERSION_CODES.O)
+	public static void scheduleSyncingProgramsForChannel(Context context, long channelId)
+	{
+		Log.d(TAG, "ProgramsRefresh job");
+		ComponentName componentName = new ComponentName(context, SyncProgramsJobService.class);
+		JobInfo.Builder builder =
+			new JobInfo.Builder(getJobIdForChannelId(channelId), componentName);
+		JobInfo.TriggerContentUri triggerContentUri =
+			new JobInfo.TriggerContentUri(
+				TvContractCompat.buildChannelUri(channelId),
+				JobInfo.TriggerContentUri.FLAG_NOTIFY_FOR_DESCENDANTS);
+		builder.addTriggerContentUri(triggerContentUri);
+		builder.setTriggerContentMaxDelay(0L);
+		builder.setTriggerContentUpdateDelay(0L);
+
+		PersistableBundle bundle = new PersistableBundle();
+		bundle.putLong(TvContractCompat.EXTRA_CHANNEL_ID, channelId);
+		builder.setExtras(bundle);
+
+		JobScheduler scheduler =
+			(JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
+		scheduler.cancel(getJobIdForChannelId(channelId));
+		scheduler.schedule(builder.build());
+	}
+
+	private static int getJobIdForChannelId(long channelId)
+	{
+		return (int) (CHANNEL_JOB_ID_OFFSET + channelId);
+	}
+
+	/**
+	 * Generates all subscriptions for homescreen channels.
+	 */
+	public static List<HomeScreenChannel> createUniversalSubscriptions()
+	{
+		List<HomeScreenChannel> subs = new ArrayList<>();
+		subs.addAll(createPlatformSubscriptions());
```

Comment:
Can me simplified with a constructor call passing the list to copy from.

Reply:
So initially I thought about adding a 'Favorites' channel that is platform agnostic, which is why it is how it is, but that would require adding that feature to the main screen then also adding a way persist this group. I figured it should be a separate PR since if its added to TV UI it should be added to the phone UI too and this PR is really just for the Oreo channels. 

I've condensed it for this PR. Can always go back and change it if a 'Favorites' row ever gets added.